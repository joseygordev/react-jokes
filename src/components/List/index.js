import React from 'react';
import PropTypes from 'prop-types';

// Components
import ListItem from '../ListItem';

const List = ({items, onClick}) => {
  return (
    <React.Fragment>
      {items.map(item => (
        <ListItem key={item.id} item={item} onClick={onClick} />
      ))}
    </React.Fragment>
  );
};

List.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      content: PropTypes.string,
    })
  ).isRequired,
  onClick: PropTypes.func
};
List.defaultProps = {
  onClick: null
};

export default List;
