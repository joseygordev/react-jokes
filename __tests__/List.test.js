import React from 'react';
import renderer from 'react-test-renderer';
import List from '../src/components/List';

describe('List', () => {
  const items = [{
    id: '1',
    title: 'My title',
    content: 'My content'
  }];

  it('Should return the List component without onClick param', () => {
    const tree = renderer
      .create(
        <List items={items} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should return the List component with onClick param', () => {
    const tree = renderer
      .create(
        <List items={items} onClick={() => {}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
