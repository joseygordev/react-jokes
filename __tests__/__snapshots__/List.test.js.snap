// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Should return the List component with onClick param 1`] = `
<button
  data-testid="button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "#FFFFFF",
      "borderLeft": "1rem solid #8a8a8a",
      "borderRadius": "0.5rem",
      "boxShadow": "0.125rem 0.125rem 0.25rem rgba(0, 0, 0, 0.25)",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "space-between",
      "margin": "8px auto",
      "padding": 16,
      "width": "100%",
    }
  }
  type="button"
>
  <h2
    style={
      Object {
        "fontSize": "1.5rem",
        "fontWeight": "bold",
        "margin": 0,
        "textTransform": "capitalize",
      }
    }
  >
    My title
  </h2>
  <p
    style={
      Object {
        "fontSize": "1.25rem",
        "margin": "8px 0px 0px",
        "textAlign": "left",
        "textTransform": "capitalize",
      }
    }
  >
    My content
  </p>
</button>
`;

exports[`List Should return the List component without onClick param 1`] = `
<div
  style={
    Object {
      "background": "#FFFFFF",
      "borderLeft": "1rem solid #8a8a8a",
      "borderRadius": "0.5rem",
      "boxShadow": "0.125rem 0.125rem 0.25rem rgba(0, 0, 0, 0.25)",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "space-between",
      "margin": "8px auto",
      "padding": 16,
      "width": "100%",
    }
  }
>
  <h2
    style={
      Object {
        "fontSize": "1.5rem",
        "fontWeight": "bold",
        "margin": 0,
        "textTransform": "capitalize",
      }
    }
  >
    My title
  </h2>
  <p
    style={
      Object {
        "fontSize": "1.25rem",
        "margin": "8px 0px 0px",
        "textAlign": "left",
        "textTransform": "capitalize",
      }
    }
  >
    My content
  </p>
</div>
`;
